packages/fast-jss-utilities/src/with-defaults.spec.ts

```diff
@@ -0,0 +1,48 @@
+import { withDefaults } from "./with-defaults";
+
+describe("withDefaults", (): void => {
+    interface WithDefaultsTest {
+        a?: string;
+        b?: string;
+    }
+
+    const testDefaults: WithDefaultsTest = {
+        a: "a",
+        b: "b",
+    };
+
+    test("should return an empty object if no arguments are passed", (): void => {
+        expect(withDefaults<WithDefaultsTest>({})({})).toEqual({});
+        expect(withDefaults<WithDefaultsTest>(void 0)({})).toEqual({});
+        expect(withDefaults<WithDefaultsTest>(void 0)(void 0)).toEqual({});
+        expect(withDefaults<WithDefaultsTest>(void 0)(undefined)).toEqual({});
+        expect(withDefaults<WithDefaultsTest>({})(void 0)).toEqual({});
+        expect(withDefaults<WithDefaultsTest>(undefined)(void 0)).toEqual({});
```

Comment:
nit: couple of these are redundant since `void 0` is the same as `undefined` in this context.